src/Identity/Core/src/DataProtectorTokenProvider.cs

```diff
@@ -110,38 +125,58 @@ public virtual async Task<bool> ValidateAsync(string purpose, string token, User
                     var expirationTime = creationTime + Options.TokenLifespan;
                     if (expirationTime < DateTimeOffset.UtcNow)
                     {
+                        Logger.LogWarning(0, "ValidateAsync failed: the expiration time is invalid.");
                         return false;
                     }
 
                     var userId = reader.ReadString();
                     var actualUserId = await manager.GetUserIdAsync(user);
                     if (userId != actualUserId)
                     {
+                        Logger.LogWarning(1, $"ValidateAsync failed: the expected and actual UserId aren't equal");
                         return false;
                     }
+
                     var purp = reader.ReadString();
                     if (!string.Equals(purp, purpose))
                     {
+                        Logger.LogWarning(2, $"ValidateAsync failed: the expected and actual purpose aren't equal.");
                         return false;
                     }
+
                     var stamp = reader.ReadString();
                     if (reader.PeekChar() != -1)
                     {
+                        Logger.LogWarning(2, $"ValidateAsync failed: unexpected end of input.");
                         return false;
                     }
 
                     if (manager.SupportsUserSecurityStamp)
                     {
-                        return stamp == await manager.GetSecurityStampAsync(user);
+                        var isEqualsSecurityStamp = stamp == await manager.GetSecurityStampAsync(user);
+                        if (!isEqualsSecurityStamp)
+                        {
+                            Logger.LogWarning(4, $"ValidateAsync failed: the expected and actual stamp aren't equal.");
+                        }
+
+                        return isEqualsSecurityStamp;
+                    }
+
+                    
+                    var stampIsEmpty = stamp == "";
+                    if (!stampIsEmpty)
+                    {
+                        Logger.LogWarning(5, "ValidateAsync failed: the expected and actual stamp aren't equal. Expected empty stamp.");
                     }
-                    return stamp == "";
+
+                    return stampIsEmpty; 
                 }
             }
-            // ReSharper disable once EmptyGeneralCatchClause
-            catch
-            {
-                // Do not leak exception
```

Comment:
@blowdart how important is it that we don't leak exceptions here?  Given the comment do we wante to err on the side of not logging the exact exception message?

Reply:
Yes, this needs to return. We cannot return the exception message, only that an exception occured.

Reply:
> Yes, this needs to return. We cannot return the exception message, only that an exception occured.

okay. Should we write a debug message when an exception occurs ? Or not ?

Reply:
No I think we should just log that an exception has occurred without any of the details in the exception, basically go back to the general catch